source/yats/include/yats/task_container.h

```diff
@@ -107,31 +107,29 @@ class task_container : public abstract_task_container
         throw std::runtime_error("A not copyable type cannot be used in multiple connections.");
     }
 
-    template <size_t index = 0, typename T = typename helper::output_type, typename Output = std::enable_if_t<std::is_same<T, void>::value, T>>
-    std::enable_if_t<(index < helper::output_count)> write(Output output)
+    template <size_t Index = 0, typename T = typename helper::output_type>
+    std::enable_if_t<(Index < helper::output_count)> write(T output)
     {
-        auto& value = std::get<index>(output);
-        const auto& following_nodes = std::get<index>(m_output);
-        for (size_t i = 0; i < following_nodes.size(); ++i)
+        // Contains the callbacks to write into inputs of the following tasks.
+        const auto& callbacks = std::get<Index>(m_output);
+        auto& slot = std::get<Index>(output);
+
+        // cend() - 1 and back() will fail for empty callbacks
```

Comment:
I think this is obvious. I would not make such a comment.

Reply:
But it does not hurt.